File: examples/governance/model_registry_domain/lib/model_registry_domain/ai_model/intended_use.rb
module ModelRegistryDomain
  class AiModel
    class IntendedUse
      attr_reader :description, :domain

      def initialize(description:, domain:)
        @description = description
        @domain = domain
        check_invariants!
        freeze
      end

      def ==(other)
        other.is_a?(self.class) &&
          description == other.description &&
          domain == other.domain
      end
      alias eql? ==

      def hash
        [self.class, description, domain].hash
      end

      private

      def check_invariants!; end
    end
  end
end

File: examples/governance_static_go/domain/intended_use.go
package domain

type IntendedUse struct {
	Description string `json:"description"`
	Domain string `json:"domain"`
}

func NewIntendedUse(description string, domain string) (IntendedUse, error) {
	v := IntendedUse{
		Description: description,
		Domain: domain,
	}
	return v, nil
}
